### PR TITLE
Add loss logging prompt-tuning

### PR DESCRIPTION
**Pull request by gkumbhat**
Supports: #179 

### Changes
- Add loss logging for prompt-tuning

### CODE REVIEW

**Review by alex-jw-brooks — APPROVED** [review]
Looks good to me!

File: caikit_nlp/modules/text_generation/peft_prompt_tuning.py [hunk 1]
```diff
@@ -1203,6 +1243,9 @@ def _execute_train_loop(
                         eval_epoch_loss,
                     )
 
+        error.value_check("<NLP66129758E>", len(training_loss_tracker) == num_epochs)
+        return {"loss": training_loss_tracker}
```
[the remark this answers is not in the document]
Comment (gkumbhat) [reply caikit_nlp/modules/text_generation/peft_prompt_tuning.py:1247]: Yep. exactly!
